apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -137,14 +142,25 @@ router.post(
         );
       }
 
+      const qid = genJobs[0]?.data['questionQid'];
+
+      const threads: AiGenerationPrompt[] = await queryRows(
+        sql.select_generation_thread_items,
+        { qid, course_id: res.locals.course.id.toString() },
+        AiGenerationPromptSchema,
+      );
+
       const result = await regenerateQuestion(
         client,
         res.locals.course.id,
         res.locals.authn_user.user_id,
-        genJobs[0]?.data?.prompt,
+        threads[0]?.user_prompt,
         req.body.prompt,
-        genJobs[0]?.data?.html,
-        genJobs[0]?.data?.python,
+        threads[threads.length - 1].html || '',
+        threads[threads.length - 1].python || '',
+        qid,
+        res.locals.user.user_id,
```

Comment:
Ditto: we want `res.locals.authn_user.user_id` here.

Reply:
changed in 8f91ed5.